src/test/java/uk/ac/cam/cl/dtg/isaac/api/managers/QuizAttemptManagerTest.java

```diff
@@ -56,12 +57,13 @@ public void fetchOrCreateWithExistingAttempt() throws AttemptCompletedException,
     assertEquals(studentAttempt, attempt);
   }
 
-  @Test(expected = AttemptCompletedException.class)
-  public void fetchOrCreateWithExistingCompletedAttemptFails()
-      throws AttemptCompletedException, SegueDatabaseException {
+  @Test
+  public void fetchOrCreateWithExistingCompletedAttemptFails() {
     withMock(quizAttemptPersistenceManager, forStudentAssignmentReturn(completedAttempt));
 
-    quizAttemptManager.fetchOrCreate(studentAssignment, student);
+    assertThrows(AttemptCompletedException.class, () -> {
```

Comment:
Not a big deal, but just in case you weren't aware, you can simplify all these even further:

```java
assertThrows(AttemptCompletedException.class, () -> quizAttemptManager.fetchOrCreate(studentAssignment, student));
```